ferveo/examples/bench_primitives_size.rs

```diff
@@ -0,0 +1,107 @@
+use ark_serialize::CanonicalSerialize;
+
+use ark_bls12_381::Bls12_381 as EllipticCurve;
+use ferveo::*;
+use ferveo_common::ExternalValidator;
+use rand::prelude::StdRng;
+use rand_core::SeedableRng;
+use std::fs::{create_dir_all, OpenOptions};
+use std::io::prelude::*;
+use std::path::Path;
+
+pub fn save_data(threshold: usize, shares_num: usize, transcript_size: usize) {
+    let dir_path = Path::new("/tmp/benchmark_setup");
+    create_dir_all(dir_path).unwrap();
+    let file_path = dir_path.join("results.md");
+
+    if !file_path.exists() {
+        eprintln!("Creating a new file: {}", file_path.display());
+        let mut file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(&file_path)
+            .unwrap();
+        writeln!(file, "|threshold|shares_num|pvss_transcript_size|",).unwrap();
+        writeln!(file, "|---|---|---|",).unwrap();
+    }
+
+    eprintln!("Appending to file: {}", file_path.display());
+    let mut file = OpenOptions::new().append(true).open(&file_path).unwrap();
+    writeln!(file, "|{}|{}|{}|", threshold, shares_num, transcript_size,)
+        .unwrap();
+}
+
+// TODO: Find a way to deduplicate the following methods with benchmarks and test setup
+
+fn gen_keypairs(num: u32) -> Vec<ferveo_common::Keypair<EllipticCurve>> {
+    let rng = &mut ark_std::test_rng();
+    (0..num)
+        .map(|_| ferveo_common::Keypair::<EllipticCurve>::new(rng))
+        .collect()
+}
+
+fn gen_validators(
+    keypairs: &[ferveo_common::Keypair<EllipticCurve>],
+) -> Vec<ExternalValidator<EllipticCurve>> {
+    (0..keypairs.len())
+        .map(|i| ExternalValidator {
+            address: format!("validator_{}", i),
+            public_key: keypairs[i].public(),
+        })
+        .collect()
+}
+
+fn setup_dkg(
+    validator: usize,
+    shares_num: u32,
+) -> PubliclyVerifiableDkg<EllipticCurve> {
+    let keypairs = gen_keypairs(shares_num);
+    let validators = gen_validators(&keypairs);
+    let me = validators[validator].clone();
+    PubliclyVerifiableDkg::new(
+        validators,
+        Params {
+            tau: 0,
+            security_threshold: shares_num / 3,
```

Comment:
Threshold being N/3 of N is not a realistic setting. We want at the very least N/2 + 1, and 2N/3 is the most likely. 
How about we parametrize this to:  N/2 + 1 (~51%), 2N/3 (~66%), 4N/5 (80%), N (100%)

Reply:
Sounds good, looking into it.